src/main/java/nextstep/courses/service/CourseService.java

```diff
@@ -1,21 +1,61 @@
 package nextstep.courses.service;
 
 import nextstep.courses.domain.Course;
-import nextstep.courses.domain.CourseRepository;
+import nextstep.courses.domain.session.Session;
+import nextstep.courses.domain.session.SessionFeeType;
+import nextstep.courses.domain.session.SessionStatus;
+import nextstep.courses.repository.CourseRepository;
+import nextstep.courses.repository.SessionRepository;
 import nextstep.users.domain.NsUser;
 import org.springframework.transaction.annotation.Transactional;
 
+import java.time.LocalDate;
+
 public class CourseService {
 
     private final CourseRepository courseRepository;
 
-    public CourseService(CourseRepository courseRepository) {
+    private final SessionRepository sessionRepository;
+
+    public CourseService(CourseRepository courseRepository, SessionRepository sessionRepository) {
         this.courseRepository = courseRepository;
+        this.sessionRepository = sessionRepository;
+    }
+
+    public Course findCourseById(Long courseId) {
+        return courseRepository
+                .findById(courseId)
+                .initSessions(sessionRepository.findByCourseId(courseId));
+    }
+
+    @Transactional
+    public void createCourse(NsUser nsUser, String title) {
+        courseRepository.save(new Course(title, nsUser.getId()));
+    }
+
+    @Transactional
+    public void createSession(
+            NsUser nsUser,
+            Long courseId,
+            String sessionTitle,
+            Long sessionNumber,
+            LocalDate startDate,
+            LocalDate endDate,
+            String url,
+            SessionFeeType sessionFeeType,
+            SessionStatus sessionStatus,
+            int capacity
+    ) {
+        Course course = findCourseById(courseId);
+        Session session = course.createSession(courseId, nsUser, sessionTitle, sessionNumber, startDate, endDate, url, sessionFeeType, sessionStatus, capacity);
+        sessionRepository.create(session);
     }
 
     @Transactional
     public void registerSession(Long courseId, Long sessionId, NsUser nsUser) {
```

Comment:
```suggestion
    public void registerSession(Long sessionId, NsUser nsUser) {
```
특정 강의에 수강신청을 하기 때문에 굳이 courseId는 필요없지 않을까?
sessionId만으로 Session을 조회하면 되지 않을까?

Reply:
한 Course가 여러 Session을 가진다고 하여, 둘의 관계를 매핑하는 것으로 Session 쪽에 courseId를 가져야한다고 생각했습니다. (현재 코드에서는 딱히 사용하지는 않지만..)
다른 요구사항이 없다면 굳이 넣지 않아도 무방하긴 합니다.
수정하겠습니다!